File: Client/src/client/systems/weather.js
/// <reference types="@altv/types-client" />
/// <reference types="@altv/types-natives" />

import alt from 'alt-client';
import native from 'natives';

alt.on('globalSyncedMetaChange', (key, value) => {
    changeWeather(key, value);
});

alt.on('connectionComplete', () => {
    let value = '';
    changeWeather('ChangeWeather', value);
});

function changeWeather(key, value) 
{
    if (key !== 'ChangeWeather') return;

    native.setWeatherTypeOvertimePersist(value, 5);
}

File: Los Angeles Life Client/Los Angeles Life Client/World/Weather.cs
using AltV.Net.Client;

namespace Los_Angeles_Life_Client.System
{
    public abstract class Weather : Client
    {
        public static void Load()
        {
            Alt.OnConnectionComplete += () =>
            {
                string? value = null;

                if (value != null) ChangeWeather("ChangeWeather", value);
            };

            Alt.OnGlobalSyncedMetaChange += OnGlobalSyncedMetaChange;
        }

        private static void ChangeWeather(string key, object value)
        {
            if (key != "ChangeWeather") return;

            Alt.Natives.SetWeatherTypeOvertimePersist(value.ToString(), 5);
        }

        private static void OnGlobalSyncedMetaChange(string key, object value, object oldValue)
        {
            ChangeWeather(key, value);
        }
    }
}